### ISSUE TITLE
False-positive Unnecessary typecast

### ISSUE THREAD

**Issue by janlanger** [kalessil/phpinspectionsea#632]
Hi, I found two false positives for unecessary typecast inspection

`(int) (round(($priceOriginal - $price) / $priceOriginal * 100 - 0.25) ?: 0)`
- probably caused by `?:`, if I try to put float after ? the false-positive disappears

```
if (!is_int($variant) && !ctype_digit($variant)) {
	return 0;
}
$variant = (int) $variant;
```
- the variant variable may still not be int after the if because of ctype_digit("1") returns true

**Comment by kalessil** [comment]
Thank you for reporting @janlanger, which IDE and plugin versions do you use?

**Comment by janlanger** [comment]
PhpStorm 2017.3 EAP Build #PS-173.3531.17

Plugin 2.3.11.1

**Comment by kalessil** [comment]
Ok, I'll check - perhaps this something we have reported already to JB.

**Comment by kalessil** [comment]
@janlanger : this is a IDE bug - https://youtrack.jetbrains.com/issue/WI-38731, closing the isse.
